### PR TITLE
LPS-162313 Add POSHI automation | Master

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://issues.liferay.com/browse/LPS-162313
cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by alessandromalves** [comment]
Hey @Tim-Cao, I didn't use `JSONLayout.addPublicLayout();` because for this test will need to activate the Hidden from Menu Display option on widget page configuration

**Comment by Tim-Cao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 1]
```diff
@@ -29,6 +29,60 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-162313. The user is able to select Hidden Pages for Web Content structure."
+	@priority = "4"
+	test AbleToSelectHiddenPagesForWebContentStructure {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:34]: SelectHiddenPageInLinkToPageField

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 2]
```diff
@@ -29,6 +29,60 @@ definition {
...
+			WebContentStructures.saveCP();
+		}
+
+		task ("When creating Web Content based on the created structure") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			WebContentNavigator.gotoAddWithStructureCP(structureName = "WC Structure Name");
+
+			WebContent.addWithStructureCP(webContentTitle = "WC WebContent Title");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:71]: Merge into one When
For example,
When the user creates a web content based on structure with Link a Page field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 3]
```diff
@@ -29,6 +29,60 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-162313. The user is able to select Hidden Pages for Web Content structure."
+	@priority = "4"
+	test AbleToSelectHiddenPagesForWebContentStructure {
+		task ("Given a Widget Page with Hidden from Menu Display option selected") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:35]: Given a user has a widget page hidden from Menu Display

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 4]
```diff
@@ -29,6 +29,60 @@ definition {
...
+
+		task ("When creating Web Content based on the created structure") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			WebContentNavigator.gotoAddWithStructureCP(structureName = "WC Structure Name");
+
+			WebContent.addWithStructureCP(webContentTitle = "WC WebContent Title");
+		}
+
+		task ("Then the page with Hidden from Menu Display option should be shown in the selection popup") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:73]: Then the user could select the hidden page in Link a Page field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 5]
```diff
@@ -29,6 +29,60 @@ definition {
...
 
+	@description = "This is a test for LPS-162313. The user is able to select Hidden Pages for Web Content structure."
+	@priority = "4"
+	test AbleToSelectHiddenPagesForWebContentStructure {
+		task ("Given a Widget Page with Hidden from Menu Display option selected") {
+			ApplicationsMenu.gotoSite(site = "Test Site Name");
+
+			ProductMenu.gotoPortlet(
+				category = "Site Builder",
+				portlet = "Pages");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:40]: PagesAdmin.openPagesAdmin

**Review by Tim-Cao — COMMENTED** [review]
Hi @alessandromalves, would you mind adapting the comments? Thanks!
Btw, please fix the sf issue due to recent Poshi syntax changes.

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by alessandromalves** [comment]
Thanks @Tim-Cao! Fixed!

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**